app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt

```diff
@@ -261,46 +282,155 @@ object GoogleMapHelper {
-            MarkerOptions().position(posMidRight).icon(
-                getMarkerRessource(
-                    R.drawable.ic_rightarrow
-                )
-            ).anchor(0.5f, 0.5f).draggable(true).snippet(PolygonAction.RIGHT.toString())
+            newMarker(
+                posMidRight,
+                0.5f,
+                0.5f,
+                PolygonAction.RIGHT.toString(),
+                null,
+                true,
+                R.drawable.ic_rightarrow,
+                0,
+                0,
+                100,
+                100,
+                100,
+                100
+            )
         )
         moveRightPos = moveRightMarker!!.position
+
         moveDownMarker = map!!.addMarker(
-            MarkerOptions().position(posMidDown).icon(
-                getMarkerRessource(
-                    R.drawable.ic_downarrow
-                )
-            ).anchor(0.5f, 0.5f).draggable(true).snippet(PolygonAction.DOWN.toString())
+            newMarker(
+                posMidDown,
+                0.5f,
+                0.5f,
+                PolygonAction.DOWN.toString(),
+                null,
+                true,
+                R.drawable.ic_downarrow,
+                0,
+                0,
+                100,
+                100,
+                100,
+                100
+            )
         )
         moveDownPos = moveDownMarker!!.position
+
         moveMarker = map!!.addMarker(
-            MarkerOptions().position(posCenter).icon(getMarkerRessource(R.drawable.ic_move))
-                .anchor(0.5f, 0.5f).draggable(true).snippet(
-                PolygonAction.MOVE.toString()
+            newMarker(
+                posCenter,
+                0.5f,
+                0.5f,
+                PolygonAction.MOVE.toString(),
+                null,
+                true,
+                R.drawable.ic_move,
+                0,
+                0,
+                100,
+                100,
+                100,
+                100
             )
         )
         movePos = moveMarker!!.position
+
+        rotationMarker = map!!.addMarker(
+            newMarker(
+                pos4,
+                0.5f,
+                0.5f,
+                PolygonAction.ROTATE.toString(),
+                null,
+                true,
+                R.drawable.ic_rotation,
+                0,
+                0,
+                100,
+                100,
+                100,
+                100
+            )
+        )
+        rotationPos = rotationMarker!!.position
+    }
+
+    /**
+     * Sets the values for the markers
+     */
+    fun newMarker(
+        pos: LatLng,
+        hAnchor: Float,
+        vAnchor: Float,
+        snippet: String?,
+        title: String?,
+        draggable: Boolean,
+        idDrawable: Int,
+        leftBound: Int,
+        topBound: Int,
+        rightBound: Int,
+        bottomBound: Int,
+        width: Int,
+        height: Int
+    ): MarkerOptions {
```

Comment:
Is it possible to group all this parameter in some kind of objects ? Because I think that the parameter list is way too long

Reply:
It should have been fixed by Mathieu's PR